pkg/executor/kubernetes.go

```diff
@@ -358,7 +362,19 @@ func (k8s *k8s) Execute(command string) (TaskHandle, error) {
 	}
 
 	// Best effort potential way to check if binary is started properly.
-	taskHandle.Wait(100 * time.Millisecond)
+	select {
+	case <-taskWatcher.failed:
+		log.Errorf("K8s executor: task %q on pod %s failed", taskHandle.command, taskHandle.podName)
+		break
+	case <-taskWatcher.deleted:
+		log.Errorf("K8s executor: pod %s deleted deleted immediately after being started", taskHandle.podName)
```

Comment:
Could you add a comment, what's the difference between deletion and failure? Is deletion a user-invoked delete?